src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorInProcess.cs

```diff
@@ -132,6 +142,242 @@ public async Task SetUseSuggestionModeAsync(bool value, CancellationToken cancel
+            var combobox = (await GetNavigationBarComboBoxesAsync(view, cancellationToken))[index];
+            return combobox.SelectedItem?.ToString();
+        }
+
+        public async Task SelectProjectNavigationBarItemAsync(string item, CancellationToken cancellationToken)
+        {
+            await SelectNavigationBarItemAsync(index: 0, item, cancellationToken);
+        }
+
+        public async Task SelectTypeNavigationBarItemAsync(string item, CancellationToken cancellationToken)
+        {
+            await SelectNavigationBarItemAsync(index: 1, item, cancellationToken);
+        }
+
+        public async Task SelectMemberNavigationBarItemAsync(string item, CancellationToken cancellationToken)
+        {
+            await SelectNavigationBarItemAsync(index: 2, item, cancellationToken);
+        }
+
+        public async Task SelectNavigationBarItemAsync(int index, string item, CancellationToken cancellationToken)
+        {
+            await TestServices.Workspace.WaitForAsyncOperationsAsync(FeatureAttribute.NavigationBar, cancellationToken);
+
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var itemIndex = await GetNavigationBarItemIndexAsync(index, item, cancellationToken);
+            if (itemIndex < 0)
+            {
+                throw new ArgumentException($"Could not find '{item}' in combobox");
+            }
+
+            await ExpandNavigationBarAsync(index, cancellationToken);
+            await TestServices.Input.SendAsync(VirtualKey.Home);
+            for (var i = 0; i < itemIndex; i++)
+            {
+                await TestServices.Input.SendAsync(VirtualKey.Down);
+            }
+
+            await TestServices.Input.SendAsync(VirtualKey.Enter);
+        }
+
+        public async Task<int> GetNavigationBarItemIndexAsync(int index, string item, CancellationToken cancellationToken)
+        {
+            var items = await GetNavigationBarItemsAsync(index, cancellationToken);
+            return items.IndexOf(item);
+        }
+
+        public async Task<bool> IsNavigationBarEnabledAsync(CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var view = await GetActiveTextViewAsync(cancellationToken);
+            return (await GetNavigationBarMarginAsync(view, cancellationToken)) is not null;
+        }
+
+        private async Task<List<ComboBox>> GetNavigationBarComboBoxesAsync(IWpfTextView textView, CancellationToken cancellationToken)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var margin = await GetNavigationBarMarginAsync(textView, cancellationToken);
+            return margin.GetFieldValue<List<ComboBox>>("_combos");
+        }
+
+        private async Task<UIElement?> GetNavigationBarMarginAsync(IWpfTextView textView, CancellationToken cancellationToken)
```

Comment:
good god.

Reply:
who do we send feedback to that VS/editor needs a proper, resilient, testing API so we can do this sort of validation without this stuff?

Reply:
Maybe @AmadeusW ?